### ISSUE TITLE
TextInput focus display issues when using clear button

### ISSUE THREAD

**Issue by mikemurray** [reactioncommerce/reaction-component-library#270]
Type: **minor**

**Describe the bug**

TextInput gets confused about its focus state resulting in the green focus ring and clear button still being visible even after blurring (un-focusing) the field.

**To Reproduce**

1. Go to https://stoic-hodgkin-c0179e.netlify.com/#!/TextInput
2. In the first example enter some text
3. The field should have a green focus ring and an "x" button now
4. Click the "x" button to clear the text
5. The field should still have a green focus outline but the cursor is no longer visible in the field
6. Click inside the field again to continue editing
7. Type in more text
8. Click outside the field onto a blank spot on the page
9. The field will still look "focused", the "x" button will still be visible

**Expected behavior**

If the field is not focused, the outline should be the default gray, and the "clear" (x) button should not be visible.

**Screenshots**

Field is still looks "focused" even though it is not

![reaction___checkout](https://user-images.githubusercontent.com/42217/45056638-61d0ae00-b048-11e8-8e74-cb645a74c722.png)

**Desktop (please complete the following information):**
 - OS: MacOS 10.13.6
 - Browser: Chrome
 - Version: 68.0.3440.106 (Official Build) (64-bit)
